### PR TITLE
fix(combo-box): support v10 input controls

### LINKED ISSUE #2240: Support v10 combobox input controls
image

image

image

image

### PR DESCRIPTION
**Pull request by emyarod**
Closes #2240

This PR adds styles for the v10 combo box selection button and refactors multiselect selection button styles

Related:

https://github.com/IBM/carbon-components-react/issues/2109

https://github.com/IBM/carbon-components-react/issues/2113

#### Changelog

**New**

- add styles for v10 combo box input controls

**Changed**

- nest warning icons within list box field

#### Testing / Reviewing

- Ensure listbox is not visually broken
- Ensure multiselect is not visually broken
- Ensure combobox is not visually broken

### CODE REVIEW

**Review by jnm2377 — APPROVED** [review]

**Review by aagonzales — COMMENTED** [review]
Looks like the invalid states are a little off. 
- Top one either missing the clear x or should be closet to the chevron
- Bottom one is missing the icons

image

**Comment by emyarod** [comment]
both points should be addressed now @aagonzales

**Comment by emyarod** [comment]
feedback from @carmacleod on the React component: https://github.com/IBM/carbon-components-react/issues/1763#issuecomment-476252889

> Some notes:
* [x] delete `tabindex="0"` from both of the outer divs, i.e. _only_ the input should take focus. So all keyboard events need to be on the input.
* [x] the `role="button"` needs to go on the div containing the triangle icon. Do not add a tabindex. Mouse events can go on this, but not keyboard events.
* [x] the `div.bx--list-box__field` should have `role="combobox"` and `aria-haspopup="listbox"`, `aria-expanded="true/false"`, and `aria-owns="unique-id-of-div.bx--list-box__menu"` (note: if using component-user-provided label for id, make sure id does not have spaces).
* [x] the `div.bx--form-item.bx--combo-box.bx--list-box` should not have any role or any ARIA related markup - it's just a generic container.
* [x] the `input` could have `role="textbox"` (which is the default for an input type="text", so technically, you can leave the role off), and it needs `aria-controls="unique-id-of-div.bx--list-box__menu"` (same id as for the aria-owns on the combobox div), `aria-autocomplete="list"` (because typing in the input filters the list), and `aria-label="Choose an item"` (delete `aria-label="ListBox input field"`). The input also needs `aria-activedescendant`, but your code there seems good.  :)
* [x] the `div.bx--list-box__menu` needs `role="listbox"` and the same aria-label as the input (i.e. `aria-label="Choose an item"`).

**Review by laurenmrice — CHANGES_REQUESTED** [review]
**Changes:**

- [x] There needs to be a 1px border bottom rule with `ui-04` color  when the dropdown is closed.
Screen Shot 2019-04-02 at 11 44 06 AM

- [x] There needs to be a 1px border bottom rule with `ui-03` color  when the dropdown is open.
Screen Shot 2019-04-02 at 11 44 43 AM

- [x] There needs to be an error field border around the examples that are showing an error state. 2px `support-01` color.

- [x] The close icon needs to be `icon-02` color. 

- [x] If you full render, all of the open dropdowns merge into eachother because you can't close them, i don't know if that was intended.

**Review by laurenmrice — CHANGES_REQUESTED** [review]
- [x] The chevron icon needs to be `icon-01`. Only the close icon needs to be `icon-02`.
Screen Shot 2019-04-02 at 2 03 34 PM

- [x] When the dropdown is open the bottom border rule of the label section needs to be `ui-03`. It is currently `ui-04`. When the dropdown is closed then the rule should be `ui-04`.

Screen Shot 2019-04-02 at 2 03 26 PM

**Comment by emyarod** [comment]
it looks like the text input styles were overriding the listbox styles there, should be fixed @laurenmrice

**Review by laurenmrice — APPROVED** [review]
Looks good 👍

File: src/components/combo-box/combo-box.hbs [hunk 1]
```diff
@@ -129,10 +247,50 @@
     </div>
   </div>
 </div>
+<div style="margin-bottom:10rem"></div>
```
Comment (asudoh) [inline src/components/combo-box/combo-box.hbs]: Instead of adding a markup for spacer, could we do a devenv-specific style?

**Review by asudoh — COMMENTED** [review]
LGTM basically, just one thing - Thanks alot @emyarod!

File: src/components/combo-box/combo-box.hbs (hunk 1 above)
Comment (emyarod) [reply src/components/combo-box/combo-box.hbs]: yes I've made that change now

**Review by asudoh — APPROVED** [review]
Thanks @emyarod for the update!

**Pull request merged**